diff --git a/packages/types/src/vocabulary/context.ts b/packages/types/src/vocabulary/context.ts
index 2fd98244c..d14fb8677 100644
--- a/packages/types/src/vocabulary/context.ts
+++ b/packages/types/src/vocabulary/context.ts
@@ -577,19 +577,54 @@ export interface SlotTable {
 	readonly 'literal.number.float': {
 		readonly exponent: unknown;
 		readonly fraction: unknown;
+		readonly imaginary: unknown;
+		readonly integer: unknown;
+		readonly marker: unknown;
+		readonly sign: unknown;
+	};
+	readonly 'literal.number.float.leading_point': {
+		readonly exponent: unknown;
+		readonly fraction: unknown;
+		readonly imaginary: unknown;
+		readonly integer: unknown;
+		readonly marker: unknown;
+		readonly sign: unknown;
+	};
+	readonly 'literal.number.float.scientific': {
+		readonly exponent: unknown;
+		readonly imaginary: unknown;
 		readonly integer: unknown;
 		readonly marker: unknown;
 		readonly sign: unknown;
 	};
 	readonly 'literal.number.integer': {
+		readonly content: unknown;
+		readonly suffix: unknown;
+	};
+	readonly 'literal.number.integer.big.binary': {
+		readonly content: unknown;
+	};
+	readonly 'literal.number.integer.big.hex': {
+		readonly content: unknown;
+	};
+	readonly 'literal.number.integer.big.octal': {
+		readonly content: unknown;
+	};
+	readonly 'literal.number.integer.binary': {
 		readonly content: unknown;
 		readonly prefix: unknown;
+		readonly suffix: unknown;
 	};
 	readonly 'literal.number.integer.hex': {
 		readonly content: unknown;
 		readonly prefix: unknown;
 		readonly suffix: unknown;
 	};
+	readonly 'literal.number.integer.octal': {
+		readonly content: unknown;
+		readonly prefix: unknown;
+		readonly suffix: unknown;
+	};
 	readonly 'literal.string': {
 		readonly content: unknown;
 		readonly contents: unknown;
@@ -2058,19 +2093,54 @@ export interface BaseContext extends GrammarContext {
 		readonly 'literal.number.float': {
 			readonly exponent: unknown;
 			readonly fraction: unknown;
+			readonly imaginary: unknown;
+			readonly integer: unknown;
+			readonly marker: string;
+			readonly sign: string;
+		};
+		readonly 'literal.number.float.leading_point': {
+			readonly exponent: unknown;
+			readonly fraction: unknown;
+			readonly imaginary: unknown;
+			readonly integer: unknown;
+			readonly marker: string;
+			readonly sign: string;
+		};
+		readonly 'literal.number.float.scientific': {
+			readonly exponent: unknown;
+			readonly imaginary: unknown;
 			readonly integer: unknown;
 			readonly marker: string;
 			readonly sign: string;
 		};
 		readonly 'literal.number.integer': {
+			readonly content: unknown;
+			readonly suffix: string;
+		};
+		readonly 'literal.number.integer.big.binary': {
+			readonly content: unknown;
+		};
+		readonly 'literal.number.integer.big.hex': {
+			readonly content: unknown;
+		};
+		readonly 'literal.number.integer.big.octal': {
+			readonly content: unknown;
+		};
+		readonly 'literal.number.integer.binary': {
 			readonly content: unknown;
 			readonly prefix: string;
+			readonly suffix: string;
 		};
 		readonly 'literal.number.integer.hex': {
 			readonly content: unknown;
 			readonly prefix: string;
 			readonly suffix: string;
 		};
+		readonly 'literal.number.integer.octal': {
+			readonly content: unknown;
+			readonly prefix: string;
+			readonly suffix: string;
+		};
 		readonly 'literal.string': {
 			readonly content: unknown | V.Literal.String.Escape<BaseContext>;
 			readonly contents:
diff --git a/packages/types/src/vocabulary/declaration.ts b/packages/types/src/vocabulary/declaration.ts
index de2d8f763..3e9454304 100644
--- a/packages/types/src/vocabulary/declaration.ts
+++ b/packages/types/src/vocabulary/declaration.ts
@@ -627,6 +627,8 @@ export namespace Declaration {
 		readonly $kind: 'declaration.variable';
 		readonly alternative?: V.Statement.Block<G>;
 		// r only
+		readonly definite?: boolean;
+		// t only
 		readonly mutable?: boolean;
 		// r only
 		readonly name?: G['slots']['declaration.variable']['name'];
diff --git a/packages/types/src/vocabulary/literal.ts b/packages/types/src/vocabulary/literal.ts
index 0fb84ee36..c908aa177 100644
--- a/packages/types/src/vocabulary/literal.ts
+++ b/packages/types/src/vocabulary/literal.ts
@@ -52,24 +52,92 @@ export namespace Literal {
 			// claimed by prt
 			readonly $kind: 'literal.number.float';
 			readonly exponent?: G['slots']['literal.number.float']['exponent'];
-			// t only
+			// pt only
 			readonly fraction?: G['slots']['literal.number.float']['fraction'];
-			// t only
+			// pt only
+			readonly imaginary?: G['slots']['literal.number.float']['imaginary'];
+			// p only
 			readonly integer?: G['slots']['literal.number.float']['integer'];
-			// t only
+			// pt only
 			readonly marker?: G['slots']['literal.number.float']['marker'];
-			// t only
+			// pt only
 			readonly sign?: G['slots']['literal.number.float']['sign'];
 			// t only
 		}
+		export namespace Float {
+			export interface LeadingPoint<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Float<G>> {
+				// claimed by pt
+				readonly $kind: 'literal.number.float.leading_point';
+				readonly exponent: G['slots']['literal.number.float.leading_point']['exponent'];
+				readonly fraction: G['slots']['literal.number.float.leading_point']['fraction'];
+				readonly imaginary?: G['slots']['literal.number.float.leading_point']['imaginary'];
+				// p only
+				readonly integer?: G['slots']['literal.number.float.leading_point']['integer'];
+				// p only
+				readonly marker: G['slots']['literal.number.float.leading_point']['marker'];
+				readonly sign?: G['slots']['literal.number.float.leading_point']['sign'];
+				// t only
+			}
+			export interface Scientific<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Float<G>> {
+				// claimed by pt
+				readonly $kind: 'literal.number.float.scientific';
+				readonly exponent: G['slots']['literal.number.float.scientific']['exponent'];
+				readonly imaginary?: G['slots']['literal.number.float.scientific']['imaginary'];
+				// p only
+				readonly integer: G['slots']['literal.number.float.scientific']['integer'];
+				readonly marker: G['slots']['literal.number.float.scientific']['marker'];
+				readonly sign?: G['slots']['literal.number.float.scientific']['sign'];
+				// t only
+			}
+			export type Any<G extends GrammarContext> =
+				| V.Literal.Number.Float<G>
+				| V.Literal.Number.Float.LeadingPoint<G>
+				| V.Literal.Number.Float.Scientific<G>;
+		}
 		export interface Integer<G extends GrammarContext> extends SubKindOf<V.Literal.Number<G>> {
 			// claimed by prt
 			readonly $kind: 'literal.number.integer';
 			readonly content?: G['slots']['literal.number.integer']['content'];
-			readonly prefix?: G['slots']['literal.number.integer']['prefix'];
-			// pt only
+			// r only
+			readonly suffix?: G['slots']['literal.number.integer']['suffix'];
+			// r only
 		}
 		export namespace Integer {
+			export interface Big<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Integer<G>> {
+				// claimed by pt
+				readonly $kind: 'literal.number.integer.big';
+			}
+			export namespace Big {
+				export interface Binary<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Integer.Big<G>> {
+					// claimed by t
+					readonly $kind: 'literal.number.integer.big.binary';
+					readonly content: G['slots']['literal.number.integer.big.binary']['content'];
+				}
+				export interface Hex<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Integer.Big<G>> {
+					// claimed by t
+					readonly $kind: 'literal.number.integer.big.hex';
+					readonly content: G['slots']['literal.number.integer.big.hex']['content'];
+				}
+				export interface Octal<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Integer.Big<G>> {
+					// claimed by t
+					readonly $kind: 'literal.number.integer.big.octal';
+					readonly content: G['slots']['literal.number.integer.big.octal']['content'];
+				}
+				export type Any<G extends GrammarContext> =
+					| V.Literal.Number.Integer.Big<G>
+					| V.Literal.Number.Integer.Big.Binary<G>
+					| V.Literal.Number.Integer.Big.Hex<G>
+					| V.Literal.Number.Integer.Big.Octal<G>;
+			}
+			export interface Binary<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Integer<G>> {
+				// claimed by prt
+				readonly $kind: 'literal.number.integer.binary';
+				readonly content: G['slots']['literal.number.integer.binary']['content'];
+				readonly prefix?: G['slots']['literal.number.integer.binary']['prefix'];
+				// pt only
+				readonly suffix?: G['slots']['literal.number.integer.binary']['suffix'];
+				// r only
+			}
 			export interface Hex<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Integer<G>> {
 				// claimed by prt
 				readonly $kind: 'literal.number.integer.hex';
@@ -79,7 +147,29 @@ export namespace Literal {
 				readonly suffix?: G['slots']['literal.number.integer.hex']['suffix'];
 				// r only
 			}
-			export type Any<G extends GrammarContext> = V.Literal.Number.Integer<G> | V.Literal.Number.Integer.Hex<G>;
+			export interface Imaginary<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Integer<G>> {
+				// claimed by p
+				readonly $kind: 'literal.number.integer.imaginary';
+			}
+			export interface Octal<G extends GrammarContext> extends SubKindOf<V.Literal.Number.Integer<G>> {
+				// claimed by prt
+				readonly $kind: 'literal.number.integer.octal';
+				readonly content: G['slots']['literal.number.integer.octal']['content'];
+				readonly prefix?: G['slots']['literal.number.integer.octal']['prefix'];
+				// pt only
+				readonly suffix?: G['slots']['literal.number.integer.octal']['suffix'];
+				// r only
+			}
+			export type Any<G extends GrammarContext> =
+				| V.Literal.Number.Integer<G>
+				| V.Literal.Number.Integer.Big<G>
+				| V.Literal.Number.Integer.Big.Binary<G>
+				| V.Literal.Number.Integer.Big.Hex<G>
+				| V.Literal.Number.Integer.Big.Octal<G>
+				| V.Literal.Number.Integer.Binary<G>
+				| V.Literal.Number.Integer.Hex<G>
+				| V.Literal.Number.Integer.Imaginary<G>
+				| V.Literal.Number.Integer.Octal<G>;
 		}
 		export interface Negative<G extends GrammarContext> extends SubKindOf<V.Literal.Number<G>> {
 			// claimed by r
@@ -89,8 +179,17 @@ export namespace Literal {
 		export type Any<G extends GrammarContext> =
 			| V.Literal.Number<G>
 			| V.Literal.Number.Float<G>
+			| V.Literal.Number.Float.LeadingPoint<G>
+			| V.Literal.Number.Float.Scientific<G>
 			| V.Literal.Number.Integer<G>
+			| V.Literal.Number.Integer.Big<G>
+			| V.Literal.Number.Integer.Big.Binary<G>
+			| V.Literal.Number.Integer.Big.Hex<G>
+			| V.Literal.Number.Integer.Big.Octal<G>
+			| V.Literal.Number.Integer.Binary<G>
 			| V.Literal.Number.Integer.Hex<G>
+			| V.Literal.Number.Integer.Imaginary<G>
+			| V.Literal.Number.Integer.Octal<G>
 			| V.Literal.Number.Negative<G>;
 	}
 	export interface Regex<G extends GrammarContext> extends SubKindOf<V.Literal<G>> {
@@ -181,8 +280,17 @@ export namespace Literal {
 		| V.Literal.Null.Undefined<G>
 		| V.Literal.Number<G>
 		| V.Literal.Number.Float<G>
+		| V.Literal.Number.Float.LeadingPoint<G>
+		| V.Literal.Number.Float.Scientific<G>
 		| V.Literal.Number.Integer<G>
+		| V.Literal.Number.Integer.Big<G>
+		| V.Literal.Number.Integer.Big.Binary<G>
+		| V.Literal.Number.Integer.Big.Hex<G>
+		| V.Literal.Number.Integer.Big.Octal<G>
+		| V.Literal.Number.Integer.Binary<G>
 		| V.Literal.Number.Integer.Hex<G>
+		| V.Literal.Number.Integer.Imaginary<G>
+		| V.Literal.Number.Integer.Octal<G>
 		| V.Literal.Number.Negative<G>
 		| V.Literal.Regex<G>
 		| V.Literal.Regex.Flags<G>
